gatewaycmd/mock.go

```diff
@@ -0,0 +1,3 @@
+package gatewaycmd
+
+//go:generate mockgen -package=mock -destination=./mocks/gateway_mock.go github.com/TykTechnologies/gateway-sdk/pkg/apim HotReloadAPI
```

Comment:
I would suggest to have //go:generate on the first line of file, but that is just personal preference.